nomad/structs/structs.go

```diff
@@ -5425,9 +5425,14 @@ type Deployment struct {
 	// JobVersion is the version of the job at which the deployment is tracking
 	JobVersion uint64
 
-	// JobModifyIndex is the modify index of the job at which the deployment is tracking
+	// JobModifyIndex is the modify index of the job at which the deployment is
+	// tracking. This is the ModifyIndex field on the job.
```

Comment:
maybe reword the comment like `This is the same as the ModifyIndex on the job` to make it clear.